adapters/globalsun/globalsun.go

```diff
@@ -0,0 +1,148 @@
+package globalsun
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type reqBodyExt struct {
+	GlobalsunBidderExt reqBodyExtBidder `json:"bidder"`
+}
+
+type reqBodyExtBidder struct {
+	Type        string `json:"type"`
+	PlacementID string `json:"placementId,omitempty"`
+	EndpointID  string `json:"endpointId,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var err error
+	adapterRequests := make([]*adapters.RequestData, 0, len(request.Imp))
+
+	for _, imp := range request.Imp {
+		reqCopy := *request
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var globalsunExt openrtb_ext.ImpExtGlobalsun
+
+		if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
+			return nil, []error{err}
+		}
+		if err = json.Unmarshal(bidderExt.Bidder, &globalsunExt); err != nil {
```

Comment:
Nitpick: given that we are only retrieving one value from the `reqCopy.Imp[0].Ext`, a small optimization can be made if we use `jsonparser` over `Unmarshal`:
```
36   func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
37       var err error
38       adapterRequests := make([]*adapters.RequestData, 0, len(request.Imp))
39   
40       for _, imp := range request.Imp {
41           reqCopy := *request
42           reqCopy.Imp = []openrtb2.Imp{imp}
43   
44 -         var bidderExt adapters.ExtImpBidder
45 -         var globalsunExt openrtb_ext.ImpExtGlobalsun
46 - 
47 -         if err = json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt); err != nil {
48 -             return nil, []error{err}
49 -         }
50 -         if err = json.Unmarshal(bidderExt.Bidder, &globalsunExt); err != nil {
51 -             return nil, []error{err}
52 -         }
   +         placementID, err := jsonparser.GetString(imp.Ext, "bidder", "placementId")
   +         if err != nil {
   +         	return nil, []error{err}
   +         }
53   
54           temp := reqBodyExt{GlobalsunBidderExt: reqBodyExtBidder{}}
55 -         temp.GlobalsunBidderExt.PlacementID = globalsunExt.PlacementID
   +         temp.GlobalsunBidderExt.PlacementID = placementID
56           temp.GlobalsunBidderExt.Type = "publisher"
57   
58           finalyImpExt, err := json.Marshal(temp)
59           if err != nil {
60               return nil, []error{err}
61           }
```